lighthouse-core/audits/dobetterweb/no-vulnerable-libraries.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audits a page to make sure there are no JS libraries with
+ * known vulnerabilities being used. Checks against a vulnerability db
+ * provided by Snyk.io and checked in locally as third-party/snyk-snapshot.json
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+
+class NoVulnerableLibrariesAudit extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Security',
+      name: 'no-vulnerable-libraries',
+      description: 'Avoids front-end JavaScript libraries'
+        + ' with known security vulnerabilities',
+      failureDescription: 'Includes front-end JavaScript libraries'
+        + ' with known security vulnerabilities',
+      helpText: 'Some third-party scripts may contain known security vulnerabilities ' +
+        ' that are easily identified and exploited by attackers.',
+      requiredArtifacts: ['JSVulnerableLibraries']
+    };
+  }
+  /**
+   * @param {object} vulns
+   * @return {string}
+   */
+  static mostSevere(vulns) {
+    vulns.map(vuln => {
+      switch (vuln.severity) {
+        case 'high':
+          vuln.numericSeverity = 3;
+          break;
+        case 'medium':
+          vuln.numericSeverity = 2;
+          break;
+        case 'low':
+          vuln.numericSeverity = 1;
+          break;
+      }
+    })
+    .sort((itemA, itemB) => itemA.numericSeverity > itemB.numericSeverity);
+
+    return vulns[0].severity.charAt(0).toUpperCase() + vulns[0].severity.slice(1);
```

Comment:
This seems a little awkward but there aren't massively better solutions. A minorly better one might be

    str.replace(/^\w/, l => l.toUpperCase())

and let's move this formatting detail down into audit() on this line: `lib.highestSeverity = this.mostSevere(lib.vulns);`

Reply:
Tis true—kind of an awkward bit no matter what. Changed and moved down.